### ISSUE TITLE
Transformer multiprocessing errors (in particular on Windows)

### ISSUE THREAD

**Issue by agoose77** [sjdv1982/seamless#16]
Error running the demo file given in `readme.md` with `seamless test_file.py`
```
IPython 6.1.0 -- An enhanced Interactive Python. Type '?' for help.
*********** ERROR in transformer .transform: execution error **************
Traceback (most recent call last):
  File "d:\pycharmprojects\seamless\seamless\core\pythreadkernel\__init__.py", line 173, in run
    self.update(self.updated, self.semaphore)
  File "d:\pycharmprojects\seamless\seamless\core\pythreadkernel\transformer.py", line 89, in update
    executor.start()
  File "c:\users\angus\appdata\local\programs\python\python36\lib\multiprocessing\process.py", line 105, in start
    self._popen = self._Popen(self)
  File "c:\users\angus\appdata\local\programs\python\python36\lib\multiprocessing\context.py", line 223, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "c:\users\angus\appdata\local\programs\python\python36\lib\multiprocessing\context.py", line 322, in _Popen
    return Popen(process_obj)
  File "c:\users\angus\appdata\local\programs\python\python36\lib\multiprocessing\popen_spawn_win32.py", line 65, in __init__
    reduction.dump(process_obj, to_child)
  File "c:\users\angus\appdata\local\programs\python\python36\lib\multiprocessing\reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: can't pickle code objects

```

**Comment by sjdv1982** [comment]
This is the second time that problems with multiprocessing show up... reverted for now to killable threads (https://github.com/sjdv1982/seamless/commit/2c09b83ccff4cec0adcda3feb25e223384b3e7a5). Could you test that it works under Windows?

**Comment by agoose77** [comment]
I will try and test today, this bug was simply caused by passing a compiled code object through a pipe

**Comment by sjdv1982** [comment]
> I will try and test today, this bug was simply caused by passing a compiled code object through a pipe
... which doesn't happen under Linux, because of os.fork()

**Comment by sjdv1982** [comment]
In the "schema" branch (which will soon be merged) I have added partial support for Windows.
commit: https://github.com/sjdv1982/seamless/commit/47e21f460ea2b85ee26317e729514c67a7970ab4

- Transformers can be executed as processes or as killable threads, as controlled by the SEAMLESS_USE_PROCESSES environment variable. This is now read on startup and applies to all transformers, per-transformer control is still todo.
- If the variable is not present, transformers are executed as processes if `os.fork()` is supported  (i.e. if `multiprocessing.get_start_method() == "fork"`. If not (i.e. on Windows), they are executed as killable threads.
- I have manually set multiprocessing.get_start_method() to "spawn" to emulate Windows. This works for a simple low-level context, but not yet for the high level (Python complains about the pickling of thread locks).
